txnsync/txngroups_test.go

```diff
@@ -363,34 +363,73 @@ func TestTxnGroupEncodingLarge(t *testing.T) {
 func BenchmarkTxnGroupEncoding(b *testing.B) {
 	txnGroups, _, _, err := txnGroupsData()
 	require.NoError(b, err)
-	var encodedGroupsBytes []byte
+	var size int
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		encodedGroupsBytes, err := encodeTransactionGroups(txnGroups, false)
+		require.NoError(b, err)
+		size = len(encodedGroupsBytes)
+		releaseEncodedTransactionGroups(encodedGroupsBytes)
+	}
+
+	fmt.Println("new data: ", size)
+}
+
+func BenchmarkTxnGroupCompression(b *testing.B) {
+	txnGroups, _, _, err := txnGroupsData()
+	require.NoError(b, err)
+	var size int
+	encodedGroupsBytes, err := encodeTransactionGroups(txnGroups, false)
```

Comment:
The `releaseEncodedTransactionGroups` should be called once per `encodeTransactionGroups`.